### ISSUE TITLE
save_best_only does not work

### ISSUE THREAD

**Issue by acegilz** [tensorflow/tensorflow#38648]
**System information**
Windows 7
Python 3.8
Tensorflow 2.1

**Describe the current behavior**

```python
filepath = "RNN_Final-{epoch:02d}-{val_accuracy:.3f}"

checkpoint = ModelCheckpoint("models\{}.hdf5".format(filepath, monitor='val_acc', verbose=1, save_best_only=True, mode='max'))
```

```python
history = model.fit(
    train_x, train_y,
    batch_size=BATCH_SIZE,
    epochs=EPOCHS,
    validation_data=(validation_x, validation_y),
    callbacks=[checkpoint],
)
```

I train the model with  _save_best_only_  enabled. It saves **all** the models in the directory, even when the val_accuracy reduces. I tried also switching to val_loss mode='min' but same behavior occurs.

**Describe the expected behavior**

I was expecting it to save only the models that had higher val / lower loss (According to definitions)
![tensorflow](https://user-images.githubusercontent.com/4284362/79604094-634bce80-80e5-11ea-8d5b-fe69262205b6.png)

**Comment by amahendrakar** [comment]
@acegilz,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here. Thanks!

**Comment by acegilz** [comment]
this was related with _val_acc_ vs _val_accuracy_

Fixed after changing all variables to _val_accuracy_
